src/JustSaying/Messaging/MessageHandling/ExactlyOnceHandler.cs

```diff
@@ -32,7 +34,7 @@ internal sealed class ExactlyOnceHandler<T> : IHandlerAsync<T> where T : Message
 
         public async Task<bool> Handle(T message)
         {
-            string lockKey = $"{message.UniqueKey()}-{_lockSuffixKeyForHandler}";
```

Comment:
So here `UniqueKey` is a virtual method that defaults to `Id.ToString()`, I've changed it so we can optionally pass a function when we register the handler, maybe not the right place to surface the api.